website/src/comment-panel.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { ReactNode, SetStateAction } from "react"
+import { useState } from "react"
+import * as Dailp from "src/graphql/dailp"
+import * as css from "./comment-panel.css"
+import { Button } from "./components"
+import { TranslatedParagraph } from "./segment"
+
+export const CommentPanel = (p: {
+  word: Dailp.FormFieldsFragment | null
+  segment: TranslatedParagraph | null
+  setCommentsPanel: React.Dispatch<SetStateAction<boolean>>
+}) => {
+  const [newCommentText, setNewCommentText] = useState<string>("")
+  const [newCommentType, setNewCommentType] = useState("")
+
+  // Prob messing up smth here
+  const [postCommentResult, postComment] = Dailp.usePostCommentMutation()
+
+  const options = [
+    // There is def a better way to do this..
+    "Story",
+    "Suggestion",
+    "Question",
+  ]
+
+  /** Call the backend GraphQL mutation. */
+  const runUpdate = async (variables: { input: Dailp.PostCommentInput }) => {
+    return await postComment(variables)
+  }
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setNewCommentText(event.target.value)
+  }
+
+  const handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setNewCommentType(event.target.value)
+  }
+
+  const handleSubmit = (e: React.MouseEvent<HTMLButtonElement>) => {
+    e.preventDefault()
+
+    runUpdate({
+      input: {
+        parentId: p.word ? p.word.id : p.segment ? p.segment.id : null,
+        parentType: p.word
+          ? Dailp.CommentParentType.Word
+          : Dailp.CommentParentType.Paragraph,
+        textContent: newCommentText,
+        commentType:
+          newCommentType == "Story"
+            ? Dailp.CommentType.Story
+            : newCommentType == "Suggestion"
+            ? Dailp.CommentType.Suggestion
+            : Dailp.CommentType.Question,
+      },
+    })
+
+    alert("Your comment has been posted!")
+    console.log("Submitted!")
+    p.setCommentsPanel(false)
+  }
+
+  return (
```

Comment:
Does our dependency `reakit` have tools to help you make this form? Reakit handles some accessibility stuff so we try to use it when we can. That being said, it can be a bit of a pain to work with, so I would love a second opinion from @CharlieMcVicker on this

Reply:
I think we could use `reakit` but in my mind, there will already be good accessibility coverage if we just put everything in a plain `form` and change the button click to an `onSubmit` on the form with `preventDefault.` To be clear, I think that the `textarea`, `label`s, `select`s and submit `button` should go in that form, but probably not your header.